### PR TITLE
Dividing state into parts

### PR DESCRIPTION
**Pull request by Kouprin**

### CODE REVIEW

File: chain/chain/src/chain.rs [hunk 1]
```diff
@@ -803,11 +803,11 @@ impl Chain {
         self.store.get_outgoing_receipts_for_shard(prev_block_hash, shard_id, last_height_included)
     }
 
-    pub fn get_state_for_shard(
+    pub fn get_state_header(
```
Comment (SkidanovAlex) [inline chain/chain/src/chain.rs]: `get_state_response_header`

File: chain/chain/src/chain.rs [hunk 2]
```diff
@@ -911,23 +907,74 @@ impl Chain {
...
             prev_chunk_header,
             prev_chunk_proof,
-            prev_payload,
             incoming_receipts_proofs,
             root_proofs,
-        ))
+        })
+    }
+
+    pub fn get_state_part(
```
Comment (SkidanovAlex) [inline chain/chain/src/chain.rs]: `get_state_response_part`

File: chain/chain/src/test_utils.rs [hunk 3]
```diff
@@ -587,17 +596,27 @@ impl RuntimeAdapter for KeyValueRuntime {
...
 
-        let (balances_root, _) = merklize(&new_balances);
-        let mut receipt_vec = state.receipt_nonces.iter().cloned().collect::<Vec<_>>();
-        receipt_vec.sort();
-        let (receipt_root, _) = merklize(&receipt_vec);
-        let mut tx_vec = state.tx_nonces.iter().cloned().collect::<Vec<_>>();
-        tx_vec.sort();
-        let (tx_root, _) = merklize(&tx_vec);
+        let data = state.try_to_vec()?;
+        let num_state_parts = 7; // TODO MOO
```
Comment (SkidanovAlex) [inline chain/chain/src/test_utils.rs]: `self.num_state_parts`?

File: chain/chain/src/chain.rs (hunk 1 above)
Comment (Kouprin) [reply chain/chain/src/chain.rs]: ++

File: chain/chain/src/chain.rs (hunk 2 above)
Comment (Kouprin) [reply chain/chain/src/chain.rs]: ++

File: chain/chain/src/test_utils.rs (hunk 3 above)
Comment (Kouprin) [reply chain/chain/src/test_utils.rs]: ++

**Pull request closed without merging**